src/tribler/core/utilities/pony_utils.py

```diff
@@ -281,27 +378,78 @@ def acquire_lock(self):
             info.lock_wait_total_duration += lock_wait_duration
 
     def release_lock(self):
+        # Adds tracking of a db_session's total lock hold duration
         super().release_lock()
         info = local.db_session_info
         if info is not None:
             acquire_time = self._acquire_time
             lock_hold_duration = time.time() - acquire_time
             info.lock_hold_total_duration += lock_hold_duration
 
+    def set_transaction_mode(self, connection, cache):
+        with marking_corrupted_db(self.pool.filename):
+            return super().set_transaction_mode(connection, cache)
+
+    def execute(self, cursor, sql, arguments=None, returning_id=False):
+        with marking_corrupted_db(self.pool.filename):
+            return super().execute(cursor, sql, arguments, returning_id)
+
+    def mark_db_as_malformed(self):
+        filename = self.pool.filename
+        if not Path(filename).exists():
+            raise RuntimeError(f'Corrupted database file not found: {filename!r}')
+
+        marker_filename = filename + '.is_corrupted'
+        Path(marker_filename).touch()
+
+    def get_pool(self, is_shared_memory_db, filename, create_db=False, **kwargs):
```

Comment:
Could you please add a description that clarifies that this method is called on each DB session and therefore the corrupted DB could be handled properly?

Reply:
And I'm curious why it is not enough to call `handle_db_if_corrupted` only in `bind` function?

Reply:
The logic was reworked entirely; the `get_pool` method was no longer necessary and is now removed.

Reply:
>  And I'm curious why it is not enough to call handle_db_if_corrupted only in bind function?

Now, the `handle_db_if_corrupted` function is not called in the `get_pool` method, as the method was removed.

`handle_db_if_corrupted` is still used in for places:
* in bind method to fix the corrupted database before connecting to it from ORM.
* in the `get_db_version` method that is called separately from ORM
* in `metadata_store/db/store.py` and in `bandwidth_accounting/db/store.py` because the current logic of these modules requires fixing the corrupted database before determining the value of the `create_db` flag.